ear-production-suite-plugins/lib/src/communication/object_metadata_sender.cpp

```diff
@@ -138,31 +138,37 @@ void ObjectMetadataSender::elevation(float value) {
 }
 void ObjectMetadataSender::distance(float value) {
   std::lock_guard<std::mutex> lock(dataMutex_);
+  data_.set_changed(true);
   data_.mutable_obj_metadata()->mutable_position()->set_distance(value);
 }
-
 void ObjectMetadataSender::width(float value) {
   std::lock_guard<std::mutex> lock(dataMutex_);
+  data_.set_changed(true);
   data_.mutable_obj_metadata()->set_width(value);
 }
 void ObjectMetadataSender::height(float value) {
   std::lock_guard<std::mutex> lock(dataMutex_);
+  data_.set_changed(true);
   data_.mutable_obj_metadata()->set_height(value);
 }
 void ObjectMetadataSender::depth(float value) {
   std::lock_guard<std::mutex> lock(dataMutex_);
+  data_.set_changed(true);
   data_.mutable_obj_metadata()->set_depth(value);
 }
 void ObjectMetadataSender::diffuse(float value) {
   std::lock_guard<std::mutex> lock(dataMutex_);
+  data_.set_changed(true);
   data_.mutable_obj_metadata()->set_diffuse(value);
 }
 void ObjectMetadataSender::factor(float value) {
   std::lock_guard<std::mutex> lock(dataMutex_);
+  data_.set_changed(true);
   data_.mutable_obj_metadata()->set_factor(value);
 }
 void ObjectMetadataSender::range(float value) {
   std::lock_guard<std::mutex> lock(dataMutex_);
+  data_.set_changed(true);
   data_.mutable_obj_metadata()->set_range(value);
 }
 
```

Comment:
I'd be a bit tempted to factor all these out into something that wraps the mutex and the set_changed call as it'd be really easy to miss either if modifying/updating this.

something like
```c++
// object_metadata_sender.hpp
class ObjectMetadataSender {
...
private:
template<typename F>
    auto setData(F f) -> decltype(f(data_)) {
        std::lock_guard<std::mutex> lock(dataMutex_);
        data_.set_changed(true);
        return f(data_.mutable_obj_metadata());
    }
};
// object_metadata_sender.cpp
void ObjectMetadataSender::range(float value) {
  setData([value](auto data){ data->set_range(value); });
}
// etc..
```
Which is a variation of [this](http://martinecker.com/martincodes/cplusplus11-wrap-pattern/). Could simplify it a bit if nothing returns a value (just have `setData` return void and drop the auto/trailing return type/decltype). Might not need the trailing type in any case if we're on C++17.

I dunno, maybe changing two repeated lines with one slightly more complicated one is overkill, what do you reckon?